fmn/rules/taskotron.py

```diff
@@ -24,5 +25,81 @@ def taskotron_task(config, message, task=None):
     if not task:
         return False
 
-    tasks = [item.strip() for item in task.split(',')]
-    return message['msg']['task'].get('name') in tasks
+    tasks = [item.strip().lower() for item in task.split(',')]
+    return message['msg']['task'].get('name').lower() in tasks
+
+
+@hint(categories=['taskotron'])
+def taskotron_changed_outcome(config, message):
+    """ Taskotron task outcome changed
+
+    With this rule, you can limit messages to only those task results
+    with changed outcomes. This is useful when an object (a build,
+    an update, etc) gets retested and either the object itself or the
+    environment changes and the task outcome is now different (e.g.
+    FAILED -> PASSED).
+    """
+
+    outcome = message['msg']['result'].get('outcome')
+    prev_outcome = message['msg']['result'].get('prev_outcome')
+
+    return outcome != prev_outcome
+
+
+@hint(categories=['taskotron'])
+def taskotron_task_outcome(config, message, outcome=None):
+    """ Particular taskotron task outcome
+
+    With this rule, you can limit messages to only those of particular
+    `taskotron <https://taskotron.fedoraproject.org/>`_ task outcome.
+
+    You can specify several outcomes by separating them with a comma ',',
+    i.e.: ``PASSED,FAILED``.
+
+    The full list of supported outcomes can be found in the libtaskotron
+    `documentation <https://docs.qadevel.cloud.fedoraproject.org/
+    libtaskotron/latest/resultyaml.html#minimal-version>`_.
+    """
+
+    if not outcome:
+        return False
+
+    outcomes = [item.strip().lower() for item in outcome.split(',')]
+    return message['msg']['result'].get('outcome').lower() in outcomes
+
+
+@hint(categories=['taskotron'])
+def taskotron_task_failed_or_changed_outcome(config, message):
```

Comment:
Can we rename this to `taskotron_task_particular_or_changed_outcome` and make it configurable similar to `taskotron_task_outcome`? Let's have `FAILED` as a value by default, but people will be able to adjust it, which will make this rule more flexible and useful. Docs need to be adjusted to say this allows any particular outcome(s) or a change outcome, and by default it only looks for `FAILED` outcomes.